kartothek/io_components/metapartition.py

```diff
@@ -1549,6 +1572,58 @@ def delete_from_store(self, dataset_uuid, store):
             store.delete(file_key)
         return self.copy(files={}, data={}, metadata={})
 
+    def get_parquet_metadata(self, store, table_name):
+        """
+        Retrieve the parquet metadata for the MetaPartition.
+        Especially relevant for calculating dataset statistics.
+
+        Parameters
+        ----------
+        store
+          A factory function providing a KeyValueStore
+        table_name
+          Name of the kartothek table for which the statistics should be retrieved
+
+        Returns
+        -------
+        pd.DataFrame
+          A DataFrame with relevant parquet metadata
+        """
+        if not isinstance(table_name, str):
+            raise TypeError("Expecting a string for parameter `table_name`.")
+
+        if callable(store):
+            store = store()
+
+        if table_name in self.files:
+            with store.open(self.files[table_name]) as fd:
+                pq_metadata = pa.parquet.ParquetFile(fd).metadata
+            data = {
+                "partition_label": self.label,
+                "number_rows_total": pq_metadata.num_rows,
+                "number_row_groups": pq_metadata.num_row_groups,
+                "serialized_size": pq_metadata.serialized_size,
+            }
+            data["row_group_id"] = range(data["number_row_groups"])
+
+            data["row_group_byte_size"], data["number_rows_per_row_group"] = zip(
+                *[
+                    (
+                        pq_metadata.row_group(rg_id).total_byte_size,
+                        pq_metadata.row_group(rg_id).num_rows,
+                    )
+                    for rg_id in data["row_group_id"]
+                ]
+            )
+            df = pd.DataFrame(data=data, columns=self._METADATA_COLUMNS,)
+            df.astype(MetaPartition._METADATA_SCHEMA._asdict())
```

Comment:
the `_asdict` looks private. isn't `dict(...)` the same thing?

Reply:
No, it isn't (but I don't know how it is different).

Reply:
> To prevent conflicts with field names, the method and attribute names start with an underscore.

Ref: https://docs.python.org/3/library/collections.html